ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -288,102 +300,86 @@ void PowerSystemNode::injectCustomFault(bool& fault_activated,
+      else
+      {
+        ROS_INFO_STREAM("Loading " << designated_file << "...");
+      }
+      if (loadCustomFaultPowerProfile(current_fault_path, designated_file))
+      {
+        custom_fault_ready = true;
+        end_fault_warning_displayed = false;
+        saved_fault_path = current_fault_path;
+        saved_file = designated_file;
+        index = 0;
+
+        ROS_WARN_STREAM_ONCE("Custom power faults may exhibit unexpected results. Caution is advised.");
+        if (designated_file == saved_file)
         {
-          ROS_INFO_STREAM("Loading " << designated_file << " and unloading " << saved_file << "...");
+          ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " reactivated!");
         }
         else
         {
-          ROS_INFO_STREAM("Loading " << designated_file << "...");
-        }
-        if (loadCustomFaultPowerProfile(current_fault_directory, designated_file))
-        {
-          custom_fault_ready = true;
-          end_fault_warning_displayed = false;
-          saved_fault_directory = current_fault_directory;
-          saved_file = designated_file;
-          index = 0;
-
-          ROS_WARN_STREAM_ONCE("Custom power faults may exhibit unexpected results. Caution is advised.");
           ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " activated!");
         }
-        else
-        {
-          // Custom fault failed to load correctly, likely due to invalid CSV formatting.
-          custom_fault_ready = false;
-        }        
       }
-      catch (int err_val)
+      else
       {
-        switch(err_val)
-        {
-          case ERR_INVALID_FILE_NAME:
-            ROS_WARN_STREAM("Invalid file name '" << designated_file << "' entered; Deactivate fault and try again.");
-            break;
-          case ERR_NO_CSV_FOUND:
-            ROS_WARN_STREAM("Could not find a CSV in the 'profiles' directory with name '" << designated_file << "'. Deactivate fault and try again.");
-            break;
-          default:
-            break;
-        }
+        // Custom fault failed to load correctly, likely due to invalid CSV formatting.
         custom_fault_ready = false;
-      }
+      }        
     }
-    else
+    catch (int err_val)
     {
-      // It's the same fault profile as previously stored, meaning the same fault was reactivated.
-      // Skip loading and continue from previous index.
-      ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " re-activated!");
-      custom_fault_ready = true;
+      switch(err_val)
+      {
+        case ERR_INVALID_FILE_NAME:
+          ROS_WARN_STREAM("Invalid file name '" << designated_file << "' entered; Deactivate fault and try again.");
+          break;
+        case ERR_NO_CSV_FOUND:
+          ROS_WARN_STREAM("Could not find a CSV in the 'profiles' directory with name '" << designated_file << "'. Deactivate fault and try again.");
+          break;
+        default:
+          break;
+      }
```

Comment:
This try-catch does not seem necessary, and I think these two errors are better handled without the use of `throw` inside `loadPowerProfile` itself, which should result in the function returning an empty `PrognoserVector`, which then propagates to `loadCustomFaultPowerProfile` returning false, which then propagates to `injectCustomFault`. And I think that works fine because all `injectCustomFault` really needs to know is whether or not the profile was successfully loaded (`true` / `false`). There's no need to have it be bogged down by the nitty gritty details of error handling that could just as easily be handled inside the function that's closer to the actual file input implementation.

try-catch blocks are useful in the right context, but here it only serves the purpose of separating file input error handling from the actual file input code.

Reply:
Just to clarify myself, I think the try-catch block inside `loadPowerProfile` that contains the while loop that iterates over file lines is good and should remain. But I feel the one inside `injectCustomFault` is unnecessary so long as `loadPowerProfile` is modified to handle those errors itself.